src/mlpack/core/data/pre_processing_impl.hpp

```diff
@@ -0,0 +1,136 @@
+#ifndef MLPACK_CORE_DATA_DATA_PRE_PROCESSING_IMPL_HPP
+#define MLPACK_CORE_DATA_DATA_PRE_PROCESSING_IMPL_HPP
+
+#include "pre_processing.hpp"
+
+namespace mlpack {
+namespace data {
+  Data::Data(std::string file_path)
+  {
+    // Constructor
+    path = file_path;
+    mlpack::data::Load(path, data, true);
```

Comment:
I see you're stripping the CSV header row, but maybe it is better to adapt `data::Load()` to use Armadillo's new(ish) functionality to read the CSV header too?  See http://arma.sourceforge.net/docs.html#save_load_mat